src/Agent.Listener/Configuration.Windows/NativeWindowsServiceHelper.cs

```diff
@@ -923,6 +937,40 @@ public void RevokeDirectoryPermissionForAccount(IList<string> folders)
             Trace.Info(StringUtil.Format($"Delete the group {groupName}."));
             DeleteLocalGroup(groupName);
         }
+        /// <summary>
+        /// Checks if account is managed service
+        /// </summary>
+        /// <param name="accountName">account name</param>
+        /// <returns>Returns true if account is managed service.</returns>
+        /// <exception cref="Win32Exception">Throws this exception if there is some error during check.</exception>
+        public bool IsManagedServiceAccount(String accountName)
+        {
+            bool isServiceAccount = false;
+            accountName = SanitizeManagedServiceAccountName(accountName);
+            var result = this.CheckNetIsServiceAccount(null, accountName, ref isServiceAccount);
+            if (result == 0)
+            {
+                return isServiceAccount;
+            }
+            else
+            {
+                int lastErrorCode = (int)GetLastError();
```

Comment:
auto lastErrorCode = Marshal.GetLastWin32Error().

Reply:
Changed to var, thanks!